skbuild/resources/cmake/UseF2PY.cmake

```diff
@@ -0,0 +1,135 @@
+#.rst:
+#
+# The following functions are defined:
+#
+# .. cmake:command:: add_f2py_target
+#
+# Create a custom rule to generate the source code for a Python extension module
+# using f2py.
+#
+#   add_f2py_target(<Name> [<F2PYInput>]
+#                   [OUTPUT_VAR <OutputVar>])
+#
+# ``<Name>`` is the name of the new target, and ``<F2PYInput>``
+# is the path to a pyf source file.  Note that, despite the name, no new
+# targets are created by this function.  Instead, see ``OUTPUT_VAR`` for
+# retrieving the path to the generated source for subsequent targets.
+#
+# If only ``<Name>`` is provided, and it ends in the ".pyf" extension, then it
+# is assumed to be the ``<F2PYInput>``.  The name of the input without the
+# extension is used as the target name.  If only ``<Name>`` is provided, and it
+# does not end in the ".pyf" extension, then the ``<F2PYInput>`` is assumed to
+# be ``<Name>.pyf``.
+#
+#
+# Options:
+#
+# ``OUTPUT_VAR <OutputVar>``
+#   Set the variable ``<OutputVar>`` in the parent scope to the path to the
+#   generated source file.  By default, ``<Name>`` is used as the output
+#   variable name.
+#
+# Defined variables:
+#
+# ``<OutputVar>``
+#   The path of the generated source file.
+#
+# Example usage
+# ^^^^^^^^^^^^^
+#
+# .. code-block:: cmake
+#
+#   find_package(F2PY)
+#
+#   # Note: In this case, either one of these arguments may be omitted; their
+#   # value would have been inferred from that of the other.
+#   add_f2py_target(f2py_code f2py_code.pyf)
+#
+#   add_library(f2py_code MODULE ${f2py_code})
+#   target_link_libraries(f2py_code ...)
```

Comment:
Do we still want to support this use case ?  Or only the case where `f2py` takes care of the compilation  ? 

Based on my last test, supporting the explicit compilation requires some more work ... https://github.com/scikit-build/scikit-build-sample-projects/tree/add-f2py-sample-projects/projects/hello-f2py#readme

Reply:
This is the only use case we need to support. I'll look into that issue after I finish this.

Reply:
This is what we're missing:

```
  copying c:\python36\lib\site-packages\numpy\f2py\src\fortranobject.c -> build\src.win32-3.6\build\src.win32-3.6\scipy\fftpack
  copying c:\python36\lib\site-packages\numpy\f2py\src\fortranobject.h -> build\src.win32-3.6\build\src.win32-3.6\scipy\fftpack
  
```

Reply:
Logged as #282.